app/components/govuk_component/task_list_component/title_component.rb

```diff
@@ -0,0 +1,37 @@
+module GovukComponent
+  class TaskListComponent::TitleComponent < GovukComponent::Base
+    attr_reader :text, :href, :hint
+
+    def initialize(text: nil, href: nil, hint: nil, classes: [], html_attributes: {})
+      @text = text
+      @href = href
+      @hint = hint
+
+      super(classes: classes, html_attributes: html_attributes)
+    end
+
+    def call
+      tag.div(**html_attributes) { title_content }
+    end
+
+  private
+
+    def title_content
+      return link if href.present?
+
+      text
+    end
+
+    def link
+      govuk_link_to(text, href, class: "govuk-task-list__link")
```

Comment:
This needs an `aria-describedby` attribute which references the status (and the hint if there is one).